src/toniarts/openkeeper/tools/convert/kmf/KmfFile.java

```diff
@@ -333,16 +332,22 @@ private List<MeshSprite> parseMeshSprites(IResourceChunkReader rawKmf, int sprsC
             sprite.setMaterialIndex(rawKmf.readUnsignedInteger());
 
             //The triangles, for each lod level
-            HashMap<Integer, List<Triangle>> trianglesPerLod = new HashMap<>(lodCount);
+            List<List<Triangle>> trianglesPerLod = new ArrayList<>(lodCount);
             for (int j = 0; j < lodCount; j++) {
-                List<Triangle> triangles = new ArrayList<>(sprite.getTriangleCounts().get(j));
-                for (int k = 0; k < sprite.getTriangleCounts().get(j); k++) {
+                int numTris = sprite.getTriangleCounts().get(j);
+                List<Triangle> triangles = new ArrayList<>(numTris);
+                for (int k = 0; k < numTris; ++k) {
                     triangles.add(new Triangle(rawKmf.readUnsignedByte(),
                             rawKmf.readUnsignedByte(),
                             rawKmf.readUnsignedByte()));
                 }
-                trianglesPerLod.put(j, triangles);
+                trianglesPerLod.add(triangles);
             }
+
+            // remove trailing empty LODs

```

Comment:
I guess that's ok looking at
https://github.com/jMonkeyEngine/jmonkeyengine/blob/76ebd1443ccc18249e9f082d528cd39baf59da84/jme3-core/src/main/java/com/jme3/scene/control/LodControl.java#L179-L183

Reply:
Gem holder looks normal in the model viewer so not sure if it breaks something there.

Reply:
Though thinking again about how that code expects a sorted set of numbers I'm not sure why it works at all, resp. doesn't fail horribly.